pkg/apis/aadpodidentity/utils.go

```diff
@@ -0,0 +1,72 @@
+package internal
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type EventType int
+
+const (
+	PodCreated      EventType = 0
+	PodDeleted      EventType = 1
+	PodUpdated      EventType = 2
+	IdentityCreated EventType = 3
+	IdentityDeleted EventType = 4
+	IdentityUpdated EventType = 5
+	BindingCreated  EventType = 6
+	BindingDeleted  EventType = 7
+	BindingUpdated  EventType = 8
+	Exit            EventType = 9
+)
+
+func IsNamespacedIdentity(azureId *AzureIdentity) bool {
+	if val, ok := azureId.Annotations[BehaviorKey]; ok {
+		if val == BehaviorNamespaced {
+			return true
+		}
+	}
+	return false
+}
+
+//TODO: is this the right place for the conversion functions?
+//TODO: V2 and internal are currently the same API. 
+
+func ConvertV1ToInternal(identityBinding *[]AzureIdentityBinding, assignedIdentity *[]AzureAssignedIdentity, identity *[]AzureIdentity) (resIdentityBinding *[]AzureIdentityBinding, resAssignedIdentity *[]AzureAssignedIdentity, resIdentity *[]AzureIdentity) {
```

Comment:
Why do we have one func for all types?

Reply:
Also needs a unit tests please.

Reply:
Moved internal api + utils file to pkg/internal
Moved conversion code into own file and created new directory pkg/conversion, added unit tests, split up into functions by type like suggested